src/pages/onboarding/Onboarding.jsx

```diff
@@ -28,26 +28,36 @@ const Onboarding = () => {
         })
         .catch((e) => setToast({ message: "Unable to load your user information. Please try again.", color: "danger" }));
     }
-  }, [isUserAuthenticated]);
+  }, [isUserAuthenticated, setToast]);
 
   useEffect(() => {
     if (isUserAuthenticated) {
       getUserInformation();
     }
   }, [isUserAuthenticated, getUserInformation, initialUsername]);
 
+  const validationErrorMessage = (msg) => {
+    setToast({ message: msg, color: "danger" });
+  };
+
   const handleUpdateUsername = () => {
     const trimmedUsername = username.trim();
     if (!trimmedUsername) {
       setToast("Username cannot be empty.");
       return;
     }
-    updateUsername(trimmedUsername).then((res) => {
-      setToast({ message: "Username updated!", color: "success" });
-      history.replace("/home");
-    }).catch((e) => {
-      setToast({ message: "Unable to update your username. Please try again.", color: "danger" });
-    });
+    if (!trimmedUsername || trimmedUsername.length < 8 || trimmedUsername.length > 15) {
+      validationErrorMessage("Your new username must be between 8 to 15 characters!");
+      return;
+    }
+    updateUsername(trimmedUsername)
+      .then((res) => {
+        setToast({ message: "Username updated!", color: "success" });
+        history.replace("/home");
+      })
+      .catch((e) => {
+        setToast({ message: e.data.description[0], color: "danger" });
```

Comment:
Is this always the format? Because otherwise, it may cause an error instead e.g. if `e.data` is null.